src/app/cli/commands/configure.ts

```diff
@@ -0,0 +1,856 @@
+        );
+
+        let message = 'Select MCP servers';
+        if (method === 'modify') {
+            message = `Select MCP servers (✓ = currently enabled, available: ${allServers.length} total)`;
+        } else if (method === 'replace') {
+            message = `Replace all MCP servers - select new ones (${allServers.length} available)`;
+        } else {
+            message = `Select MCP servers (${allServers.length} available)`;
+        }
+
+        // Show current state before selection
+        if (method === 'modify' && currentServerIds.length > 0) {
+            p.note(`Currently enabled: ${currentServerIds.join(', ')}`, 'Current MCP Servers');
+        }
+
+        const selectedIds = await p.multiselect({
+            message,
+            options: allServers.map((server) => {
+                const isCurrentlyEnabled = currentServerIds.includes(server.id);
+                return {
+                    value: server.id,
+                    label: `${server.name}${isCurrentlyEnabled ? ' ✓' : ''}`,
+                    hint: `${server.category} - ${server.description}`,
+                };
+            }),
+            initialValues: method === 'replace' ? [] : currentServerIds,
+            required: false,
+        });
+
+        // Debug the selection result
+        logger.debug(
+            `MCP Selection Result: selectedIds=${Array.isArray(selectedIds) ? selectedIds.join(', ') : 'invalid'}`
+        );
+
+        // Handle cancellation or empty selection
+        if (!selectedIds || !Array.isArray(selectedIds)) {
+            logger.warn('MCP server selection cancelled or invalid, keeping existing servers');
+            return baseServers || {};
+        }
+
+        for (const serverId of selectedIds as string[]) {
+            const server = MCP_SERVER_REGISTRY[serverId];
+            if (server) {
+                selectedServers[server.id] = server.config;
+            } else {
+                logger.warn(`Selected MCP server '${serverId}' not found in registry`);
+            }
+        }
+    }
+
+    return selectedServers;
+}
+
+/**
+ * Get preset server configurations
+ */
+function getPresetServers(preset: string): Record<string, McpServerConfig> {
+    const presets = {
+        essential: ['filesystem', 'puppeteer'],
+        developer: ['filesystem', 'puppeteer', 'github', 'terminal'],
+        productivity: ['filesystem', 'puppeteer', 'notion', 'slack'],
+        data: ['filesystem', 'sqlite', 'postgres'],
+    };
+
+    const serverIds = presets[preset as keyof typeof presets] || [];
+    const servers: Record<string, McpServerConfig> = {};
+
+    for (const id of serverIds) {
+        const server = MCP_SERVER_REGISTRY[id];
+        if (server) {
+            servers[id] = server.config;
+        }
+    }
+
+    return servers;
+}
+
+/**
+ * Configure system prompt
+ */
+async function configureSystemPrompt(basePrompt?: string | object): Promise<string | object> {
```

Comment:
i like this idea, it's a good start

i think we can take this a lot further in future PRs with some good templates

will think about how we can make this experience better for all sub-parts of each config